service/controller/v1/resource_set.go

```diff
@@ -104,6 +106,24 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 		}
 	}
 
+	var patchFinalizerResource controller.Resource
+	{
+		c := patchfinalizer.Config{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+		}
+
+		patchFinalizerResource, err = patchfinalizer.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		err := patchFinalizerResource.EnsureCreated(context.Background(), nil)
```

Comment:
Why not add it to the resource list? That way it gets run multiple times if the finalizer change fails or something.

Reply:
We have CPs without tenant clusters but want to run the patching anyway.